optuna/integration/lightgbm_tuner/optimize.py

```diff
@@ -457,6 +448,45 @@ def best_params(self) -> Dict[str, Any]:
             params.update(self.lgbm_params)
             return params
 
+    @property
+    def best_booster(self) -> lgb.Booster:
```

Comment:
Looking at the new logic, this shouldn't really be a `@property` as operations aren't trivial. It's probably good to keep backwards compatibility but I'd personally deprecate this one and replace it with a method.

Reply:
I think it's okay to discuss it later as we already have an approval for the current changes.

Reply:
> Looking at the new logic, this shouldn't really be a @property as operations aren't trivial. 

I agree with you. This behavior is too complicated as a property.
So, I'll create a `get_best_booster` method and deprecate the `best_booster` property in this PR.
@smly What do you think?

Reply:
Totally agree with your proposed change. 👍

Reply:
Thank you for your quick response. I'll apply changes.

Reply:
I added the `LightGBMTuner.get_best_booster` method and deprecated `LightGBMTuner.best_booster` in https://github.com/optuna/optuna/pull/1076/commits/bc6d1a1e94be9627b520f30b334d153facddcb77.

Reply:
Thanks for the discussion and the changes!